b2/sync/folder.py

```diff
@@ -157,6 +163,10 @@ def _walk_relative_paths(self, local_dir, b2_dir, reporter):
                 )
             local_path = os.path.join(local_dir, name)
             b2_path = join_b2_path(b2_dir, name)
+
+            if policies_manager.exclude(local_path):
```

Comment:
The check should be on the B2 name, not the local path.  Here's a test that fails because of this:

```
    def test_exclusion_with_exact_match(self):
        expected_list = [
            six.u('.dot_file'),
            six.u('hello.'),
            six.u('hello/a/1'),
            six.u('hello/a/2'),
            six.u('hello/b'),
            six.u('inner/a.bin'),
            six.u('inner/a.txt'),
            six.u('inner/b.bin'),
            six.u('inner/b.txt'),
            six.u('inner/more/a.bin'),
            six.u('inner/more/a.txt'),
            six.u('\u81ea\u7531'),
        ]
        polices_manager = ScanPoliciesManager(
            exclude_file_regexes=['hello0'],
        )
        self._check_file_filters_results(polices_manager, expected_list)
```